src/test/scala/linting/rule/LintConflictingModuleNamesSpec.scala

```diff
@@ -0,0 +1,189 @@
+}
+
+case class UnstableNameAnnotation(target: IsModule) extends SingleTargetAnnotation[IsModule] {
+  def duplicate(newTarget: IsModule): UnstableNameAnnotation = this.copy(target = newTarget)
+}
+
+class RenameDesiredNamesSpec extends FirrtlPropSpec with FirrtlMatchers {
+  val transform = new RenameDesiredNames
+
+  case class TestCase(
+    input: String,
+    annos: Seq[Annotation]
+  )
+
+  def stabilizeNames(testCase: TestCase): CircuitState = {
+    stabilizeNames(CircuitState(Parser.parse(testCase.input), UnknownForm, testCase.annos))
+  }
+
+  def stabilizeNames(state: CircuitState): CircuitState = {
+    transform.runTransform(state)
+  }
+
+  def test(testCases: TestCase *): Unit = {
+    val firstState = stabilizeNames(testCases.head)
+    val firstUnstableNames = firstState.annotations.collect {
+      case a: UnstableNameAnnotation => a
+    }
+    val firstStableNames = firstState.annotations.collect {
+      case a: StableNameAnnotation => a
+    }
+
+    testCases.tail.foldLeft(firstStableNames, firstUnstableNames) {
+      case ((stable, unstable), testCase) =>
+        val state = stabilizeNames(testCase)
+        val currUnstableNames = state.annotations.collect {
+          case a: UnstableNameAnnotation => a
+        }
+        val currStableNames = state.annotations.collect {
+          case a: StableNameAnnotation => a
+        }
+
+        currStableNames should be (stable)
+        currUnstableNames should not be (unstable)
+        (currStableNames, currUnstableNames)
+    }
+  }
+
+  property("It should rename modules if it can") {
+    val top = CircuitTarget("Foo")
+    val testCase = TestCase(
+      """|circuit Foo:
+         |  module Bar_1:
+         |    output in1: UInt<1>
+         |  module Bar_2:
+         |    output in1: UInt<1>
+         |    output in2: UInt<1>
+         |  module Bar_3:
+         |    output in1: UInt<1>
+         |    output in2: UInt<1>
+         |  module Foo:
+         |    inst bar_1 of Bar_1
+         |    inst bar_2 of Bar_2
+         |    inst bar_3 of Bar_3
+         |""".stripMargin,
+      Seq(
+        DesiredNameAnnotation("Bar_1", top.module("Bar_1")),
+        OverrideDesiredNameAnnotation("BarWith1Input", top.module("Bar_1")),
+
+        // these renames should fail because they conflict
+        OverrideDesiredNameAnnotation("BarWith2Inputs", top.module("Bar_2")),
+        OverrideDesiredNameAnnotation("BarWith2Inputs", top.module("Bar_3"))
+      )
+    )
+    val outputState = stabilizeNames(testCase)
+    val check =
+      """|circuit Foo:
+         |  module BarWith1Input:
+         |    output in1: UInt<1>
+         |  module Bar_2:
+         |    output in1: UInt<1>
+         |    output in2: UInt<1>
+         |  module Bar_3:
+         |    output in1: UInt<1>
+         |    output in2: UInt<1>
+         |  module Foo:
+         |    inst bar_1 of BarWith1Input
+         |    inst bar_2 of Bar_2
```

Comment:
Shouldn't the result be

` inst bar_2 of BarWith2Inputs
  inst bar_3 of BarWIth2Inputs_1
`
or something? Silently ignoring this and reverting back to whatever the module's desired name was doesn't make sense.

Reply:
so we should error if we can't rename? or try renaming anyway? I thought the ask was to not do any renaming in the case of module conflicts, and let the linting pass handle any errors. If there is some other naming behavior we want then I think we should bring back the configurable naming strategy thing. and if we want to error in this transform I don't understand the point of splitting the error handling into a separate lint pass.

Reply:
I guess then I am confused by the comment "        // these renames should fail because they conflict"

What does "fail" mean? Be ignored?

Reply:
ignored